packages/next/src/client/app-index.tsx

```diff
@@ -169,11 +169,19 @@ const StrictModeIfEnabled = process.env.__NEXT_STRICT_MODE_APP
   ? React.StrictMode
   : React.Fragment
 
+let unpatchConsoleError: (() => void) | undefined = undefined
 function Root({ children }: React.PropsWithChildren<{}>): React.ReactElement {
   React.useEffect(() => {
     if (process.env.__NEXT_ANALYTICS_ID) {
       require('./performance-relayer-app')()
     }
+
+    if (process.env.NODE_ENV !== 'production') {
+      // Ensure the console.error patch that collects hydration errors is reverted after hydration is finished.
+      if (unpatchConsoleError) {
+        unpatchConsoleError()
+      }
+    }
```

Comment:
unfortunately, we never know when hydration is finally finished. Today hydration is done 1 boundary at a time. So the root can commit and run effects before all suspense boundaries are hydrated. This is somewhat necessary because the server may still be streaming in suspense children while react is already attempting to hydrate the app. If React SSR does not specifically instruct the React client to client render an unresolved Suspense it will wait to hydrate it until the SSR content is streamed in which technically can take forever.

Since this is Dev only, I'd suggest we just leave it patched and not worry about cleaning up

Reply:
Ah yeah that makes sense, I'll remove it. Thanks!